sdk/azcore/runtime/pager.go

```diff
@@ -0,0 +1,95 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package runtime
```

Comment:
I placed this in the `runtime` package as customers never need to create these.  That said, I know that pagers (and pollers) are core concepts, so I could add a type alias in the `azcore` package if we think there's value exposing the type there (the constructor and supporting types would remain in `runtime`).

Reply:
The current Go 1.18 beta doesn't support aliasing a generic type.  Unclear if this will land in 1.18 or not.

Reply:
It looks like we won't have type-aliasing until Go 1.19.  I personally don't feel this is a blocking issue, and we can add the type-alias later which is non-breaking.

Reply:
I just tried this in some code I'm working on with Go 1.18 beta 2:
type Q = collections.Queue[int]   // Type alias to a generic type

var q Q

and this compiles just fine. What do you mean by type-aliasing of a generic isn't in 1.18 beta 2?

Reply:
In `azcore`, I'd like to declare `type Pager = runtime.Pager` which is not yet supported.

Your example is an alias to a generic instantiation which is slightly different.